src/EnsembleKalmanInversions.jl

```diff
@@ -238,7 +239,7 @@ end
 # it's not adaptive
 adaptive_step_parameters(::Nothing, Xⁿ, Gⁿ, y, Γy, process) = step_parameters(X, G, y, Γy, process)
 
-function step_parameters(eki::EnsembleKalmanInversion, convergence_rate)
+function step_parameters(eki::EnsembleKalmanInversion, convergence_rate, adaptive_step_parameters)
```

Comment:
I didn't quite grasp the logic here. This implements a feature that allows users to pass a function `adaptive_step_parameters` that overrides the `adaptive_step_parameters` in the source?

Should we use multiple dispatch to control the behavior of `adaptive_step_parameters` instead? This would mean generalizing the `convergence_rate` parameter (which is now either `nothing` or `Float64`) to something more general (eg `adaptivity`...)

If we stick with this design, then I guess we will find ourselves providing alternative `adaptive_step_parameters` functions that users can import into their scripts, and pass into `iterate`. But those new functions will ignore the property `convergence_rate` (and may require closing any additional parameters of those alternative functions... ?)  That might get a little tricky.

As is, without any alternatives, only advanced users who are ready to rewrite functions like https://github.com/CliMA/ParameterEstimocean.jl/blob/3dd370b774f7d415288f61f51efd3ac7aa1b5d0b/src/EnsembleKalmanInversions.jl#L285 can use this.

Reply:
Yes, this was intended to be a temporary hack. I'm all for implementing a good time-stepping API in the next PR. I'll go ahead and merge.